dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -0,0 +1,175 @@
+        // height
+        size[0] = volume(sample_shape.begin(), sample_shape.begin() + axis);
+        // width
+        size[1] = sample_shape[ndim - has_channels - 1];
+        int row_stride = sample_shape[ndim - has_channels - 1] * num_channels;
+        auto* window_gpu = window_tmp_buffer_gpu + i * kWindowCopyBufferSize;
+        args.push_back(SampleArguments{
+            size,                                              // Input matrix dimensions
+            static_cast<int>(window.tensor_shape_span(i)[0]),  // Window sizes
+            num_channels,                                      // channels count (innermost)
+            {in.tensor_data(i), row_stride},                   // Tensor-ref for source matrix A
+            window_gpu,                                        // Pointers to windows
+            {out.tensor_data(i), row_stride},                  // Tensor-ref for source matrix C
+            {out.tensor_data(i), row_stride},  // Tensor-ref for destination matrix D
+            {scale, 0}                         // Scalars used in the Epilogue
+        });
+      }
+    } else {
+      // Outer or not-innermost - repack arguments
+      for (int i = 0; i < num_samples; i++) {
+        cutlass::Array<int, 2> size;
+        auto sample_shape = in.tensor_shape(i);
+        // height
+        size[0] = sample_shape[axis];
+        // width
+        size[1] = volume(sample_shape.begin() + axis + 1, sample_shape.end());
+        auto strides = GetStrides(sample_shape);
+        int row_stride = strides[axis];
+        int planes = volume(sample_shape.begin(), sample_shape.begin() + axis);
+        int plane_stride = axis > 0 ? strides[axis - 1] : 0;
+        auto* window_gpu = window_tmp_buffer_gpu + i * kWindowCopyBufferSize;
+        args.push_back(SampleArguments{
+            size,                                              // Input matrix dimensions
+            static_cast<int>(window.tensor_shape_span(i)[0]),  // Window sizes
+            1,                                 // channels don't matter for outer dimensions
+            {in.tensor_data(i), row_stride},   // Tensor-ref for source matrix A
+            window_gpu,                        // Pointers to windows
+            {out.tensor_data(i), row_stride},  // Tensor-ref for source matrix C
+            {out.tensor_data(i), row_stride},  // Tensor-ref for destination matrix D
+            {scale, 0},                        // Scalars used in the Epilogue
+            planes,
+            plane_stride});
+      }
+    }
+    // Construct and invoke the CUTLASS kernel
+    CutlassConv gemm_operator;
+    auto status = gemm_operator.can_implement(args);
+    DALI_ENFORCE(status == cutlass::Status::kSuccess,
+                 make_string("Operation not possible: ", cutlass::cutlassGetStatusString(status)));
+    gemm_operator(args, ctx.gpu.stream);
+  }
+
+ private:
+  static constexpr bool kInnerConv = axis == ndim - has_channels - 1;
```

Comment:
The lack of "is" in the name brought me here, looking for what this name means.

Reply:
Done